doc/media-settings/Media-based-Port-settings.md

```diff
@@ -0,0 +1,313 @@
+
+              Below example shows the media-based settings for a device on a Broadcom based platform.  For each port the list of supported optics and DAC types are defined (uniquely identified by media type key). For each media type the list of supported vendors (uniquely identified by vendor key) is defined. Each vendor key which will in turn contain the key value pairs per lane that needs to be programmed.
+
+The media type key defined in the file will be comprised of compliance code and length string or simply compliance code if length does not apply (e.g. 40GBASE-CR4-3M or 40GBASE-SR4).
+
+When a media is detected, the port on which it is detected is identified at the first level. At the second level, the media type key is constructed based on compliance code and length. If compliance code -- length key is not found in the media types listed for the port, then the key is reduced to include only compliance code. For instance, if the initial media type key constructed from the EEPROM fields is '40GBASE-CR4-2M' and the specific port does not list such an entry then key is reduced to '40GBASE-CR4' and the port entry is searched if the reduced key exists. If there is still no match, then the 'Default' value listed under the port is selected.
+
+At the third level, the Vendor key is derived by concatenating vendor name and vendor part number (e.g. DELL-00-11-22).  If there is no exact match for vendor name -- vendor part number key, then the vendor key is reduced to vendor name alone (e.g. DELL) and the media type entry is looked up for such a key. If there is no match, then the default value listed under the media type is chosen. Below is an example for json file for a specific port. For the port Ethernet20 two specific media types (40GBASE-SR4 and 40GBASE-CR4-3M) and a 'Default' media type is defined. For 40GBASE-SR4 a DELL vendor name- vendor PN specific setting is listed along with a 'Default' value which is chosen for all other 40GBASE-SR4 optics apart from the listed key (DELL-00-11-22). For 40GBASE-CR4-3M media key a DELL vendor name alone key word is listed along with a 'Default' key word. This would imply that any 40GBASE-CR4-3M DAC manufactured by DELL will pick the values listed in the vendor name key whereas everything else will pick the default values.
+
+{
+
+    "PORT_OPTICS_SETTINGS": {
+
+        "Ethernet20": {
+
+            "Default": {
+
+                "Default": {
+
+                    "preemphasis": [
+
+                         "0x1201",
+
+                          "0x1234"
+
+                    ],
+
+                    "idriver": [
+
+                        "0x1",
+
+                        "0x1"
+
+                     ],
+
+                     "ipredriver": [
+
+                        "0x1",
+
+                        "0x1"
+
+                     ]
```

Comment:
Did you think about adding a FEC setting here (for copper transceivers)?

Reply:
FEC can already be provided in port_config.ini (e.g. wnc/x86_64-wnc_osw1800-r0/OSW1800-48x6q/port_config.ini). FEC settings should be coming from config DB through the configuration present in port_config.ini

Reply:
correct me if I'm wrong but is not the FEC set in the INTERFACE section, at least that is how we do it for specific optics
    "Ethernet36": {
...
      "fec": "rs"
    },

Reply:
> correct me if I'm wrong but is not the FEC set in the INTERFACE section, at least that is how we do it for specific optics
> "Ethernet36": {
> ...
> "fec": "rs"
> },

Hi. What i meant to convey here is there are multiple other mechanisms used to configure FEC and hence this specific mechanism need not have provision to configure FEC settings